docs/concepts/gardenlet.md

```diff
@@ -1,119 +1,137 @@
-Gardener used its single controller-manager which was responsible for all shoot clusters on all seed clusters, and it was performing its reconciliation loops centrally from the garden cluster.
-
-While this works well at scale for thousands of clusters today, our goal is to enable true scalability following the Kubernetes principles (beyond the capacity of a single controller-manager):
-We have now worked on distributing the logic (or the Gardener operator) into the seed cluster and introduced a corresponding component, adequately named the **Gardenlet**.
-It is Gardener's primary "agent" on every seed cluster and is only responsible for shoot clusters located in its particular seed cluster.
+Gardener is implemented using the [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/): It uses a custom `controller-manager` that acts on our own custom resources, and applies Kubernetes principles to manage clusters instead of containers. Following this analogy, you can recognize components of the Gardener architecture as well-known Kubernetes components, for example, shoot clusters can be compared with pods, and seed clusters can be seen as worker nodes.
+
+The following Gardener components play a similar role as the corresponding components in the Kubernetes architecture:
+
+| Gardener Component | Kubernetes Component | 
+|:---|:---|
+| `gardener-scheduler` | `kube-scheduler` |
+| gardenlet | kubelet |
+
+Similar to how the `kube-scheduler` of Kubernetes finds an appropriate node for newly created pods, the `gardener-scheduler` of Gardener finds an appropriate seed cluster to host the control plane for newly ordered clusters. By providing multiple seed clusters for a region or provider, and distributing the workload, Gardener also reduces the blast radius of potential issues.
+
+Last, but not least, Kubernetes runs a primary "agent" on every node, the kubelet, which is responsible for managing pods and containers on its particular node. Decentralizing the responsibility to the kubelet has the advantage that the overall system is scalable. Gardener achieves the same for cluster management by using a **gardenlet** as primary "agent" on every seed cluster, and is only responsible for shoot clusters located in its particular seed cluster:
 
 ![](gardenlet-architecture-similarities.png)
 
-The gardener-controller-manager still kept its control loops for other resources of the Gardener API, however, it does no longer talk to seed/shoot clusters.
+The `gardener-controller-manager` has control loops to manage resources of the Gardener API. However, instead of letting the `gardener-control-manager` talk directly to seed clusters or shoot clusters, the responsibility isn’t only delegated to the gardenlet, but also managed using a reversed control flow: It's up to the gardenlet to contact the Gardener API server, for example, to share a status for its managed seed clusters. 
```

Comment:
```suggestion
The `gardener-controller-manager` has control loops to manage resources of the Gardener API. However, instead of letting the `gardener-controller-manager` talk directly to seed clusters or shoot clusters, the responsibility isn’t only delegated to the gardenlet, but also managed using a reversed control flow: It's up to the gardenlet to contact the Gardener API server, for example, to share a status for its managed seed clusters. 
```